chat-plugins/mafia.js

```
@@ -0,0 +1,610 @@
+		this.destroy();
+	}
+
+	playerWindow(image, content) {
+		this.sendRoom('|html|' + this.game.mafiaWindow(image, content));
+	}
+
+	getRole() {
+		this.sendRoom('|html|' + this.game.mafiaWindow(this.class.image, this.class.flavorText));
+	}
+
+	targetWindow(image, content) {
+		let output = content;
+		output += '<br/><p>Who do you wish to target?</p>';
+		for (let i in this.validTargets) {
+			output += '<button value="/mafia target ' + this.validTargets[i].userid + '" name="send">' + this.validTargets[i].name + '</button>';
+		}
+
+		this.sendRoom('|uhtml|mafia' + this.game.room.gameNumber + 'target' + this.game.gamestate + this.game.day + '|' + this.game.mafiaWindow(image, output));
+	}
+
+	updateTarget(image) {
+		this.sendRoom('|uhtmlchange|mafia' + this.game.room.gameNumber + 'target' + this.game.gamestate + this.game.day + '|' + this.game.mafiaWindow(image, 'Targeting ' + this.target.name + '!'));
+	}
+
+	voteWindow(image, content) {
+		let output = content;
+		output += '<br/><p>Who do you wish to vote for?</p>';
+		for (let i in this.validVotes) {
+			output += '<button value="/mafia vote ' + this.validVotes[i].userid + '" name="send">' + this.validVotes[i].name + '</button>';
+		}
+
+		console.log("Showing shit to " + this.name);
+		this.sendRoom('|uhtml|mafia' + this.game.room.gameNumber + 'vote' + this.game.gamestate + this.game.day + '|' + this.game.mafiaWindow(image, output));
+	}
+
+	// Targeting mechanics:
+
+	// Targets a single player of side side.
+	singleTarget(side, targetSelf) {
+		this.validTargets = {};
+		for (let i in this.game.players) {
+			let thisSide = this.game.players[i].class.side;
+			if ((side === 'all' || thisSide === side || (side === 'nomafia' && thisSide !== 'mafia')) && (targetSelf || this.game.players[i] !== this)) {
+				this.validTargets[i] = this.game.players[i];
+			}
+		}
+	}
+
+	// Triggers after the user has selected their target.
+	onReceiveTarget(target) {
+		if (!this.targeting) {
+			return;
```

This is a silent failure for  a direct command